Sources/Swift/Core/Tools/ViewCapture/SentryUIRedactBuilder.swift

```diff
@@ -9,138 +10,270 @@ import WebKit
+            self.ignoreClassesIdentifiers.insert(ExtendedClassIdentifier(class: type))
         }
         
         for type in options.maskedViewClasses {
-            self.redactClassesIdentifiers.insert(ObjectIdentifier(type))
+            self.redactClassesIdentifiers.insert(ExtendedClassIdentifier(class: type))
         }
     }
     
+    /// Returns `true` if the provided class type is contained in the ignore list.
+    ///
+    /// This compares by string description to avoid touching Objective‑C class objects directly.
     func containsIgnoreClass(_ ignoreClass: AnyClass) -> Bool {
-        return ignoreClassesIdentifiers.contains(ObjectIdentifier(ignoreClass))
+        /// Edge case: ``UITextField`` uses an internal type of ``UITextFieldLabel`` for the placeholder, which should also be ignored
+        if ignoreClass.description() == "UITextFieldLabel" {
+            return ignoreClassesIdentifiers.contains(where: { $0.classId == "UITextField" })
+        }
+        return ignoreClassesIdentifiers.contains(where: { $0.classId == ignoreClass.description() })
     }
     
-    func containsRedactClass(_ redactClass: AnyClass) -> Bool {
-        var currentClass: AnyClass? = redactClass
-        while currentClass != nil && currentClass != UIView.self {
-            if let currentClass = currentClass, redactClassesIdentifiers.contains(ObjectIdentifier(currentClass)) {
+    /// Returns `true` if the view class (and, when required, the backing layer class) matches
+    /// one of the configured redact identifiers.
+    ///
+    /// - Parameters:
+    ///   - viewClass: Concrete runtime class of the `UIView` instance under inspection.
+    ///   - layerClass: Concrete runtime class of the view's backing `CALayer`.
+    ///
+    /// Matching rules:
+    /// - We traverse the view class hierarchy to honor base‑class entries (e.g. matching `UILabel` for subclasses).
+    /// - If an identifier specifies a `layerId`, the layer’s type description must match as well.
+    ///
+    /// Examples:
+    /// - A custom label `class MyTitleLabel: UILabel {}` will match because `UILabel` is in the redact set:
+    ///   `containsRedactClass(viewClass: MyTitleLabel.self, layerClass: CALayer.self) == true`.
+    /// - SwiftUI image drawing: `viewClass == SwiftUI._UIGraphicsView` and `layerClass == SwiftUI.ImageLayer`
+    ///   will match because we register `("SwiftUI._UIGraphicsView", layerId: "SwiftUI.ImageLayer")`.
+    /// - SwiftUI structural background: `viewClass == SwiftUI._UIGraphicsView` with a generic `CALayer`
+    ///   will NOT match (no `ImageLayer`), so we don’t redact background fills.
+    /// - `UIImageView` will match the class rule; the final decision is refined by `shouldRedact(imageView:)`.
+    func containsRedactClass(viewClass: AnyClass, layerClass: AnyClass) -> Bool {
+        var currentClass: AnyClass? = viewClass
+        while let iteratorClass = currentClass {
```

Comment:
`m`: An idea for speeding this up maybe for later: When there is a class that we should redact and it's not in the `redactClassesIdentifiers` we could add it to the `redactClassesIdentifiers`, so we don't have to traverse the classes up every time. I'm not sure if it's worth the effort, and I'm also unsure how often this code gets called.